lib/screen/main/main_view_state.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'main_view_state.freezed.dart';
+
+@freezed
+class MainViewState with _$MainViewState {
+  const factory MainViewState.init() = _Init;
+
+  const factory MainViewState.error(String message) = _Error;
+}
```

Comment:
Why don't we use MainViewState for case loading and success to display indicator and list? 

```suggestion
class MainViewState with _$MainViewState {
  const factory MainViewState.init() = _Init;

  const factory MainViewState.loading() = _Loading;

  const factory MainViewState.success(
      List<ProductModel>? products) = _Success;
      
  const factory MainViewState.error(String message) = _Error;
}
```

Reply:
@ducbm051291 I tried to avoid handling widgets from `MainViewState` because if we handle showing a widget in `MainViewState` we have to handle another widget in each state as well something like this 👇 

![Screen Shot 2566-07-13 at 17 26 52](https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/28002315/8d8e59b8-02c7-4d3a-be40-6f2e623f98a2)

I think the current implementation is flexible. and FYI: We can consider observing by `Consumer` instead in the future(maybe task for showing error and I will work on it) so let's keep it like this for now 🙇‍♀️

Reply:
Ok, thanks for clarifying